ledger/test-common/test-certificates/BUILD.bazel

```diff
@@ -5,37 +5,138 @@ load("@os_info//:os_info.bzl", "is_windows")
+chmod 444 $(location server.crt)
 
-# Generate client key, csr and crt
+# Generate Client CA private key
 $(location @openssl_dev_env//:openssl) genrsa -out $(location client.key) 4096
 $(location @openssl_dev_env//:openssl) pkey -in $(location client.key) -out $(location client.pem)
-$(location @openssl_dev_env//:openssl) req -new -key $(location client.key) -out $(location client.csr) -subj '/CN=0.0.0.0.client'
-$(location @openssl_dev_env//:openssl) x509 -req -in $(location client.csr) -CA $(location ca.crt) -CAkey $(location ca.key) -CAcreateserial -out $(location client.crt) -extfile $(location openssl-extensions.cnf) -extensions req_ext -days 3650
+$(location @openssl_dev_env//:openssl) req -new -key $(location client.pem) \
+    -subj '/CN=0.0.0.0.client' \
+    -addext "subjectAltName = DNS:localhost, IP:127.0.0.1" \
+    -out $(location client.csr)
+# Sign Client Cert
+$(location @openssl_dev_env//:openssl) ca -batch -config $(location openssl.cnf) \
+    -extensions usr_cert -notext -md sha256 \
+    -in $(location client.csr) \
+    -out $(location client.crt)
+# Validate cert is correct
+$(location @openssl_dev_env//:openssl) verify -CAfile $(location ca.crt) \
+    $(location client.crt)
+
+
+# Generate OCSP Server private key
+$(location @openssl_dev_env//:openssl) genrsa \
+    -out $(location ocsp.key.pem) 4096
+# Sign OCSP Server certificate
+$(location @openssl_dev_env//:openssl) req -config $(location openssl.cnf) -new -sha256 \
+    -subj '/CN=ocsp.127.0.0.1' \
+    -key $(location ocsp.key.pem) \
+    -out $(location ocsp.csr)
+$(location @openssl_dev_env//:openssl) ca -batch -config $(location openssl.cnf) \
+    -extensions ocsp -days 375 -notext -md sha256 \
+    -in $(location ocsp.csr) \
+    -out $(location ocsp.crt)
+# Validate extensions
+$(location @openssl_dev_env//:openssl) x509 -noout -text \
+    -in $(location ocsp.crt)
+
+
+# Generate Client-Revoked CA private key
+$(location @openssl_dev_env//:openssl) genpkey -out $(location client-revoked.key) -algorithm RSA -pkeyopt rsa_keygen_bits:2048
+$(location @openssl_dev_env//:openssl) pkey -in $(location client-revoked.key) -out $(location client-revoked.pem)
+$(location @openssl_dev_env//:openssl) req -new -key $(location client-revoked.pem) \
+    -subj '/CN=0.0.0.0.clientrevoked' \
+    -addext "subjectAltName = DNS:localhost, IP:127.0.0.1" \
+    -out $(location client-revoked.csr)
+# Sign Client Cert
+$(location @openssl_dev_env//:openssl) ca -batch -config $(location openssl.cnf) \
+    -extensions usr_cert -notext -md sha256 \
+    -in $(location client-revoked.csr) \
+    -out $(location client-revoked.crt)
+# Validate cert is correct
+$(location @openssl_dev_env//:openssl) verify -CAfile $(location ca.crt) \
+    $(location client-revoked.crt)
+# Revoke
+$(location @openssl_dev_env//:openssl) ca -batch -config $(location openssl.cnf) -revoke $(location client-revoked.crt)
+
+
+# Following is a simple-smoke test for the generated certificates
```

Comment:
I'm not convinced we need this smoke test as part of the generation of the certs.

Reply:
Let's remove that part then 🗑️